### PR TITLE
doc: add more context for the witness feature limitation

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
Add more context for the witness feature limitation.
Please help to check!

### CODE REVIEW

File: docs/advanced/witness.md [hunk 1]
```diff
@@ -56,6 +56,14 @@ The general upgrade requirements and procedures apply to clusters with a witness
 
 The witness node does not store any data so only two replicas are created for each Longhorn volume in the cluster. However, the default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as **Degraded** on the Longhorn UI.
 
+The above case only happens when you use the three-node cluster, which means this cluster does not contain any worker nodes. The Longhorn replica could be scheduled for either the worker or management nodes.
```
Comment (jillian-maroket) [inline docs/advanced/witness.md:59]:
```suggestion
The witness node does not store any data. This means that in three-node clusters (no worker nodes), only two replicas are created for each Longhorn volume. However, the default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as **Degraded** on the Longhorn UI.
```

Comment (jillian-maroket) [reply docs/advanced/witness.md:59]: It is better to add the new information to the existing paragraph.

File: docs/advanced/witness.md [hunk 2]
```diff
@@ -56,6 +56,14 @@ The general upgrade requirements and procedures apply to clusters with a witness
...
+
+:::note
+It can be summarized as follows:
+- Three-node cluster (two management nodes and one witness node): You need to create a new StorageClass with the replica count set to `2`.
+- More than three nodes cluster: No need to do anything.
+:::
+
 To resolve this issue, create a new *default* StorageClass with the **Number of Replicas** parameter set to **2**. This ensures that only two replicas are created for each Longhorn volume.
```
Comment (jillian-maroket) [inline docs/advanced/witness.md:67]:
```suggestion
In summary, you must use a StorageClass that matches the cluster configuration.

- 2 management nodes + 1 witness node: Create a new default StorageClass with the **Number of Replicas** parameter set to **2**. This ensures that only two replicas are created for each Longhorn volume.
- 2 management nodes + 1 witness node + 1 or more worker nodes: You can use the existing default StorageClass.
```

**Review by jillian-maroket — COMMENTED** [review]
@Vicente-Cheng I reorganized the information in the **Limitations** section. Please also update line 19:

> Harvester v1.3.0 supports clusters with two management nodes and one witness node.

New wording: Harvester v1.3.0 supports clusters with two management nodes and one witness node (and optionally, one or more worker nodes).

**Comment by Vicente-Cheng** [comment]
Hi @jillian-maroket,
All suggestions are updated.

Also, I add a new section to describe the relationship between replica numbers and nodes.
Could you help to recheck?

Once that's done, I will copy it to the v1.3 version.
Thanks!

File: docs/advanced/witness.md [hunk 3]
```diff
@@ -52,11 +52,18 @@ The witness node only runs the following essential workloads:
 
 The general upgrade requirements and procedures apply to clusters with a witness node. However, the existence of degraded volumes in such clusters may cause upgrade operations to fail. 
 
+## Longhorn Replicas with a Witness Node
+
+The Longhorn storage system is a highly avaiable distributed storage system that create replicas for volumes. Longhorn is provisioned to management or worker nodes, but not to witness nodes. The default replica number is set to 3. That means in following scenarios, 3 nodes cluster with 1 witness node, Longhorn will not be able to create the default number of replicas. This will cause the potential risk for high availability. Please check on the following section to get more details.
```
Comment (jillian-maroket) [inline docs/advanced/witness.md:57]:
```suggestion
## Longhorn Replicas in Clusters with a Witness Node

Harvester uses Longhorn, a distributed block storage system, for management of block device volumes. Longhorn is provisioned to management and worker nodes but not to witness nodes, which do not store any data.

Longhorn creates replicas of each volume to increase availability. Replicas contain a chain of snapshots of the volume, with each snapshot storing the change from a previous snapshot. In Harvester, the default StorageClass `harvester-longhorn` has a replica count value of `3`.
```

File: docs/advanced/witness.md [hunk 4]
```diff
@@ -52,11 +52,18 @@ The witness node only runs the following essential workloads:
...
+
 ## Limitations
 
-The witness node does not store any data so only two replicas are created for each Longhorn volume in the cluster. However, the default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as **Degraded** on the Longhorn UI.
+The default StorageClass `harvester-longhorn` of Harvester has a replica count value of `3` for high availability. According to above, if you use this StorageClass to create volumes with 2 nodes + 1 witness node, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as **Degraded** on the Longhorn UI. If your cluster have more than 3 nodes, the volume will be created successfully and **healthy**.
```
Comment (jillian-maroket) [inline docs/advanced/witness.md:61]:
```suggestion
Witness nodes do not store any data. This means that in three-node clusters (no worker nodes), only two replicas are created for each Longhorn volume. However, the default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as **Degraded** on the Longhorn UI.
```

**Review by jillian-maroket — COMMENTED** [review]
@Vicente-Cheng I don't understand why you changed what I wrote for the Limitations section and also mentioned the same information in the section that you added. Each section must have a distinct purpose so I rewrote your draft to match what I think you're trying to achieve.

Also, please apply ALL of the changes to the v1.3.0 doc.

**Comment by Vicente-Cheng** [comment]
Hi @jillian-maroket, 

Sorry, I misunderstood what you meant with
> It is better to add the new information to the existing paragraph.

So, I try to add a new section and make it clearer.
Thanks for revising this again.

Update with all suggestions!

**Comment by jillian-maroket** [comment]
@Vicente-Cheng Please apply all of the changes to the v1.3.0 doc so that I can finally approve. Thanks!

**Comment by Vicente-Cheng** [comment]
Hi @jillian-maroket,

Updated to versioned v1.3 docs.
Please help to check. Thanks!

**Review by jillian-maroket — APPROVED** [review]
LGTM. Thanks!

**Review by bk201 — APPROVED** [review]

**Pull request merged**